src/server.js

```diff
@@ -253,10 +264,26 @@ export class Server extends EventEmitter {
 
     handlerRun(handler) {
         this.handlers[handler.id] = handler;
+
         handler.once('destroy', () => {
```

Comment:
Is handler for 'destroy' necessary? 'handlerClosed' will be called anyway, right?

Reply:
I cleaned it up a little, but had to keep the destroy event. It would take longer time to figure out why it's needed, but some of the tests were failing when I moved all of the functionality to  handlerClose and removed destroy event. Will take a look at it later.